src/main/java/com/infoshareacademy/service/ReadFileService.java

```diff
@@ -0,0 +1,73 @@
+package com.infoshareacademy.service;
+
+import com.google.gson.Gson;
+
+import com.google.gson.reflect.TypeToken;
+import com.infoshareacademy.model.Categories;
+import com.infoshareacademy.model.Questions;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.lang.reflect.Type;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class ReadFileService {
+
+    private static Path pathQuestions = Path.of(System.getProperty("user.dir"),"Questions.txt");
+
+    public static List<Questions> loadQuestions(Categories categories) throws IOException {
+        List<Questions> allQuestions = null;
+        try {
+            // create Reader
+            Reader reader = Files.newBufferedReader(pathQuestions);
+            System.out.println(pathQuestions);
+            System.out.println(reader);
+            // create Gson instance
+            Gson gson = new Gson();
+            // for converting JSON array to a List, we need to use TypeToken class
+            Type userListType = new TypeToken<ArrayList<Questions>>(){}.getType();
+            // get data from JSON file and convert to Question class
+            allQuestions = gson.fromJson(reader, userListType);
```

Comment:
```suggestion
           List<Questions> allQuestions = gson.fromJson(reader, userListType);
```
Wtedy nie potrzeba inicjalizować listy na początku metody

Reply:
gdy tak zrobię to inicjalizacja jest wewnątrz "try'a" i jest problem żeby zwrócić to na koniec metody. Zmusza mnie by  inicjalizacja była poza try'em na początku metody.

Reply:
OK, w takim razie możesz zainicjalizować zmienną na początku metody
List<Questions> allQuestions = new ArrayList<>();
Dzięki temu nie będzie latał po aplikacji null.